File: seabattle-server/src/main/java/com/example/seabattleserver/model/BooleanResponse.java
package com.example.seabattleserver.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public record BooleanResponse(@JsonProperty boolean response, @JsonProperty String errorMessage) {

    public static BooleanResponse of(boolean value) {
        return new BooleanResponse(value, null);
    }

    public static BooleanResponse error(String message) {
        return new BooleanResponse(false, message);
    }
}

File: SeaBattle/app/src/main/java/com/example/seabattle/api/model/BooleanResponse.kt
package com.example.seabattle.api.model

data class BooleanResponse(
    private var response: Boolean? = null,
    private var errorMessage: String? = null
) {
    public fun getResponse(): Boolean {
        if (response == null) {
            return false
        }
        return response as Boolean
    }

    public fun getMessage(): String {
        if (errorMessage == null) {
            return ""
        }
        return errorMessage as String
    }
}